File: elemental/src/dto/commentodto.ts
import {UserDTO} from "./userdto";
import {PostDTO} from "./postdto";

export class CommentoDTO {
  id: number

  nomeAutore: string

  contenuto: string

  dataCommento: Date

  user: UserDTO

  post: PostDTO

  constructor(id: number,

  nomeAutore: string,

  contenuto: string,

  dataCommento: Date,

  user: UserDTO,

  post: PostDTO) {
    this.id = id
    this.nomeAutore = nomeAutore
    this.contenuto = contenuto
    this.dataCommento = dataCommento
    this.user = user
    this.post = post
  }
}

File: elemental-service/src/main/java/com/simonecampis/Elemental/dto/CommentoDTO.java
package com.simonecampis.Elemental.dto;

import com.simonecampis.Elemental.model.Post;
import com.simonecampis.Elemental.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CommentoDTO {

    private Long id;

    private String nomeAutore;

    private String contenuto;

    private Date dataCommento;

    private User user;

    private Post post;

}
